src/signals/incident/containers/IncidentContainer/reducer.js

```diff
@@ -110,16 +108,27 @@ export default (state = initialState, action) => {
         ]
       }
 
+      if (
+        selection.length >
+        action.payload[action.payload.meta_name].maxNumberOfAssets
+      ) {
+        maxAssetWarning = true
+        selection = previousSelection
+        location = state.get('incident').toJS().location
+      }
+
       return state.set(
         'incident',
         fromJS({
           ...state.get('incident').toJS(),
           ...{
             [action.payload.meta_name]: {
               selection,
-              location: action.payload.location,
+              location,
             },
           },
+          ...{ location },
```

Comment:
omdat we immutable gebruiken is het niet nodig om location en maxAssetWarning de dekstructuren.

Reply:
weet niet of dat Nederlands is ;)

Reply:
haha, dekstructereren misschien niet, maar destructureren misschien wel?
ik heb het nog wat opgeschoond in ieder geval zodat het er wat eenvoudiger uitziet.